app/models/pipeline_run.rb

```diff
@@ -1019,10 +1041,7 @@ def compile_stats_file
     end
 
     # Load unidentified reads
-    unidentified = all_counts.detect { |entry| entry.value?("unidentified_fasta") }
-    if unidentified
-      self.unmapped_reads = unidentified[:reads_after]
-    end
+    self.unmapped_reads = fetch_unmapped_reads(all_counts) || unmapped_reads
```

Comment:
what does `|| unmapped_reads` do here?

Reply:
it keeps the current value if `fetch_unmapped_reads` returns nil because of an error

Reply:
interesting. Makes sense - so `fetch_unmapped_reads  -> nil` is both the "not yet available" case (for which `|| unmapped_reads` doesn't change anything) and the "error" case (for which `|| unmapped_reads` does do something)